pkg/apis/serving/v1alpha1/revision_validation.go

```diff
@@ -143,6 +144,9 @@ func validateContainer(container corev1.Container) *apis.FieldError {
 	if err := validateProbe(container.LivenessProbe).ViaField("livenessProbe"); err != nil {
 		errs = errs.Also(err)
 	}
+	if _, err := name.ParseReference(container.Image, name.WeakValidation); err != nil {
+		errs = errs.Also(apis.ErrInvalidValue(container.Image, "image"))
```

Comment:
I'm worried that this isn't indicating _**why**_ the name is invalid...

Reply:
I've included the returned error message, but that's not particularly useful still.

Please don't git blame [the TODO](https://github.com/google/go-containerregistry/blob/1d7c260b3d3c1b33ff96c1ec002b70db42f7d877/pkg/name/ref.go#L49), but in general we should improve the error messages coming out of the ggcr image reference parsing stuff. Even if we included the errors returned by `name.NewFoo`, I don't think it would be any more helpful than just "this is wrong".

See the error from the linked issue:
```
repository can only contain the runes `abcdefghijklmnopqrstuvwxyz0123456789_-./`: knative-samples/autoscale-go:0.1".
```

I think that's not great :/